Competition2017/src/competition/subsystems/drive/commands/CalculateDistanceOfBoilerParallelCommand.java

```diff
@@ -0,0 +1,71 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import com.google.inject.Inject;
+
+import competition.subsystems.pose.PoseSubsystem;
+import competition.subsystems.vision.DetectedBoiler;
+import competition.subsystems.vision.VisionSubsystem;
+import xbot.common.command.BaseCommand;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class CalculateDistanceOfBoilerParallelCommand extends BaseCommand{
+    private double angleOfLineFromRobotToBoiler;
+    private PoseSubsystem pose;
+    private VisionSubsystem vision;
+    private double distanceOnParallel;
+    private DriveForDistanceCommand driveForDistance;
+    private DoubleSupplier parallelAngleSupplier;
+
+    private boolean isFinished = false;
+    
+    @Inject
+    public CalculateDistanceOfBoilerParallelCommand(
+            PoseSubsystem pose,
+            VisionSubsystem vision,
+            XPropertyManager propMan){
+        this.pose = pose;
+        this.vision = vision;
+    }
+    
+    public void setParallelAngle(double angle) {
+        parallelAngleSupplier = () -> angle;
+    }
+    
+    public void setParallelAngleProp(DoubleProperty prop) {
+        parallelAngleSupplier = () -> prop.get();
+    }
+
+    @Override
+    public void initialize() {
```

Comment:
Most of our commands include an "initializing" to help us track runtime flow. I'd recommend it here and in other new commands.